code/GatewayFieldsFactory.php

```diff
@@ -23,14 +23,44 @@ class GatewayFieldsFactory
     protected $gateway;
     protected $groupdatefields = true;
 
+    protected static $renamemap = array();
```

Comment:
Should not be static, every instance of `GatewayFieldsFactory` needs to have its own map.